### PR TITLE
Add base geoscanner implementation

### PR DESCRIPTION
**Pull request by SirEdvin**
So, this is base implementation for geoscanner block logic. This pull request is not yet ready, please, check a list of todo

- [x] Add textures
- [x] Add crafting receipt
- [x] Consider some burst scanner logic with RF cost
- [x] Configurable max free scan radius and max cost scan radius
- [x] Add scan delay logic
- [x] Something else?
- [x] Check api from #92 
- [x] Remove `deepScan` method
- [x] Add `chunkAnalyze` method

### CODE REVIEW

File: src/main/java/de/srendi/advancedperipherals/common/addons/computercraft/base/BaseTurtle.java [hunk 1]
```diff
@@ -62,8 +62,8 @@ public void update(@NotNull ITurtleAccess turtle, @NotNull TurtleSide side) {
         if (!turtle.getWorld().isClientSide) {
             IPeripheral turtlePeripheral = turtle.getPeripheral(side);
             this.turtle = turtle;
-            if (turtlePeripheral instanceof EnvironmentDetectorPeripheral)
-                ((EnvironmentDetectorPeripheral) turtlePeripheral).setTurtle(turtle);
+            if (turtlePeripheral instanceof BasePeripheral)
```
[the remark this answers is not in the document]
Comment (Seniorendi) [reply src/main/java/de/srendi/advancedperipherals/common/addons/computercraft/base/BaseTurtle.java:65]: This step is quite important. See https://github.com/SquidDev-CC/CC-Tweaked/discussions/796

**Comment by Seniorendi** [comment]
So. Now i have time for you ^^
I really love your work, thank you for the contribution. Everything looks good, I will now test the geo scanner

**Comment by SirEdvin** [comment]
@Seniorendi Just to be on same like, I will resolve conflicts today/tommorow and also will remove `deepScan` method (because it is mostly useless as we have block reader) and will add method `chunkScanner` that will return amount or ores and materials in this chunk. 

Is this okay?)

**Comment by Seniorendi** [comment]
> @Seniorendi Just to be on same like, I will resolve conflicts today/tommorow and also will remove `deepScan` method (because it is mostly useless as we have block reader) and will add method `chunkScanner` that will return amount or ores and materials in this chunk.
> 
> Is this okay?)

Yes this isn't a problem

**Comment by SirEdvin** [comment]
@Seniorendi sorry for delay, merge request is fully ready!

**Review by Seniorendi — APPROVED** [review]

**Pull request merged**